camel/storages/graph_storages/nebula_graph.py

```diff
@@ -188,32 +189,188 @@ def ensure_edge_type_exists(
         self,
         edge_type: str,
     ) -> None:
-        r"""Ensures that a specified edge type exists in the NebulaGraph
-        database. If the edge type already exists, this method does nothing.
+        """Creates an edge type with temporal support if it doesn't exist.
 
-        Args:
-            edge_type (str): The name of the edge type to be created.
+        Each edge type will have a timestamp property to track temporal
+        information.
 
-        Raises:
-            Exception: If the edge type creation fails after multiple retry
-                attempts, an exception is raised with the error message.
+        Args:
+            edge_type (str): The type of edge to be created.
         """
-        create_edge_stmt = f'CREATE EDGE IF NOT EXISTS {edge_type}()'
+        create_edge_stmt = (
+            f'CREATE EDGE IF NOT EXISTS {edge_type}' '(timestamp timestamp)'
+        )
 
         for attempt in range(MAX_RETRIES):
             res = self.query(create_edge_stmt)
             if res.is_succeeded():
-                return  # Tag creation succeeded, exit the method
+                return
 
             if attempt < MAX_RETRIES - 1:
                 time.sleep(RETRY_DELAY)
             else:
-                # Final attempt failed, raise an exception
                 raise Exception(
-                    f"Failed to create tag `{edge_type}` after "
+                    f"Failed to create edge {edge_type} after "
                     f"{MAX_RETRIES} attempts: {res.error_msg()}"
                 )
 
+    def add_temporal_triplet(
+        self,
+        subj: str,
+        obj: str,
+        rel: str,
+        timestamp: Optional[datetime] = None,
+    ) -> None:
+        """Adds a temporal triplet (subject-relationship-object with
+           timestamp).
+
+        Args:
+            subj (str): The identifier for the subject entity.
+            obj (str): The identifier for the object entity.
+            rel (str): The identifier for the relationship type.
+            timestamp (datetime, optional): The timestamp for the relationship.
+                Defaults to current time if not specified.
+        """
+        self.ensure_tag_exists(subj)
+        self.ensure_tag_exists(obj)
+        self.ensure_edge_type_exists(rel)
+        self.add_node(node_id=subj, tag_name=subj)
+        self.add_node(node_id=obj, tag_name=obj)
+
+        timestamp = timestamp or datetime.now()
+        formatted_timestamp = timestamp.strftime("%Y-%m-%d %H:%M:%S")
+
+        insert_stmt = (
+            f'INSERT EDGE IF NOT EXISTS {rel}(timestamp) VALUES '
+            f'"{subj}"->"{obj}":('
+            f'timestamp("{formatted_timestamp}"))'
+        )
+
+        res = self.query(insert_stmt)
+        if not res.is_succeeded():
+            raise Exception(
+                f'Failed to create temporal relationship: {res.error_msg()}'
+            )
```

Comment:
please try this func by real graph data

Reply:
got it, commit not ready now. re-request review will be setted once tests are done